openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2242,6 +2477,59 @@ def get_unitless_values(attr):
                     )
                 )
 
+    @staticmethod
+    def _check_partial_bond_orders_from_molecules_duplicates(pb_mols):
+        if len(set(map(Molecule.to_smiles, pb_mols))) < len(pb_mols):
+            raise ValueError(
+                "At least two user-provided fractional bond order "
+                "molecules are isomorphic"
+            )
+
+    @staticmethod
+    def _assign_partial_bond_orders_from_molecules(topology, pbo_mols):
```

Comment:
I've moved these two methods to the base class since they're used by both BondHandler and ProperTorsionHandler. I thought about keeping them as functions at the module level, but their use is so specifically tied to the behavior of ParameterHandlers that I think they fit best as a staticmethod of the class.